### PR TITLE
Doc: Add Product Editor Extensibility Guidelines

### LINKED ISSUE #40565: Update developer handbook with extensibility guidelines
## Description

We should create a new page on the handbook based on the extensibility guidelines posted by the design team.

Extensibility guidelines post: peCask-rP-p2

It should contain only the subset that is actually implemented, as the post also contains parts that are still under development. It should be a separate page from the root README, adding links to the root README where appropriate.

### PR DESCRIPTION
**Pull request by mattsherman**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR supersedes #40635 and adds the initial version of the product editor extensibility guidelines, based on the latest draft document.

Closes #40565.

### How to test the changes in this Pull Request:

1. Read docs and verify no spelling-mistakes or terminology issues.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: docs/product-editor-development/product-editor-extensibility-guidelines.md [hunk 1]
```diff
@@ -0,0 +1,236 @@
+# Product Editor Extensibility Guidelines
+
+> ⚠️ **Notice:** These guidelines are currently a **work in progress**. Please be aware that some details might be incomplete or subject to change. We appreciate your patience and welcome any contributions!
+
+Changelog:
+
+- 2021-11-17: Initial version
```
Comment (nathanss) [inline docs/product-editor-development/product-editor-extensibility-guidelines.md]: Is that correct? That long ago?

File: docs/product-editor-development/product-editor-extensibility-guidelines.md [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+
+## Form Extensibility
+
+**Our research shows that merchants think in terms of tasks and goals while creating new products in Woo.** For example, adding or editing a product price is a separate task from uploading images or managing inventory.
+
+For this reason, the new product form features groups (tabs) that correspond with these tasks. Simply speaking, they are separate views where different form features live. As a developer, you can extend any of these views and offer merchants the information and tools they need to create better, more successful products.
+
+There are several ways to extend the new product form: from a single field to a whole group or section containing multiple fields and tables. These extension points are tightly linked to the form structure, giving you plenty of freedom to create a great user experience.
```
Comment (nathanss) [inline docs/product-editor-development/product-editor-extensibility-guidelines.md]: I just wondered if the word 'tightly' in 'tightly linked' is serving any purpose here. It almost sounds like a bad thing. Maybe just 'linked' would be enough? Your call :)

**Review by nathanss — APPROVED** [review]
Left some comments, but I'm pre-approving! Finally we're wrapping this up...

File: docs/product-editor-development/product-editor-extensibility-guidelines.md (hunk 1 above)
Comment (mattsherman) [reply docs/product-editor-development/product-editor-extensibility-guidelines.md]: Hahaha... yes, we've been sitting on this for a few years... oops!!! Thanks for catching this!

File: docs/product-editor-development/product-editor-extensibility-guidelines.md (hunk 2 above)
Comment (mattsherman) [reply docs/product-editor-development/product-editor-extensibility-guidelines.md]: I agree, I'll remove "tightly". Good call.

**Pull request merged**